apps/insights/src/app/api/pyth/get-pyth-feeds-demo-data/[datasource]/[symbol]/route.ts

```diff
@@ -0,0 +1,85 @@
+import { isNullOrUndefined } from "@pythnetwork/shared-lib/util";
+import { NextRequest, NextResponse } from "next/server";
+
+import { fetchHistoricalDataForPythFeedsDemo } from "../../../../../../pyth-feed-demo-data/fetch-historical-data-from-db";
+import {
+  ALLOWED_EQUITY_SYMBOLS,
+  DATA_SOURCES_REPLAY,
+  ValidDateSchema,
+} from "../../../../../../schemas/pyth/pyth-pro-demo-schema";
+
+export const GET = async (
+  req: NextRequest,
+  ctx: { params: Promise<{ datasource: string; symbol: string }> },
+) => {
+  const { datasource, symbol } = await ctx.params;
+  const datasourceValidation =
+    await DATA_SOURCES_REPLAY.safeParseAsync(datasource);
+
+  if (datasourceValidation.error) {
+    return NextResponse.json(
+      { error: datasourceValidation.error.message },
+      { status: 400 },
+    );
+  }
+
+  const symbolValidation = await ALLOWED_EQUITY_SYMBOLS.safeParseAsync(symbol);
```

Comment:
There's no reason to use async parsing here, you only need async zod parsing if you're using async transforms